python3/libexec/probe-device-for-file

```diff
@@ -40,11 +42,13 @@ def stderr_redirected(to=os.devnull):
                                             # CLOEXEC may be different
 
 if __name__ == "__main__":
+    device = None
+    file =  None
     if len(sys.argv) != 3:
-       print("Usage: %s <device> <file>" % sys.argv[0])
-       sys.exit(2)
-    device = sys.argv[1]
-    file = sys.argv[2]
+        print("Usage: %s <device> <file>" % sys.argv[0])
+        sys.exit(2)
+        device = sys.argv[1]
+        file = sys.argv[2]
```

Comment:
@ashwin9390 Please be aware, that in Python, when we're moving the indentation, we change the *meaning* of the code and with it, it's logic.

In this case, when you moved the indentation of
```py
    device = sys.argv[1]
    file = sys.argv[2]
```
by 4 columns to the right, you removed it from the main instruction stream and into the if-clause for handling the case when not arguments were given on the command line.
```py
    if len(sys.argv) != 3:
        print("Usage: %s <device> <file>" % sys.argv[0])
        sys.exit(2)
        device = sys.argv[1]
        file = sys.argv[2]
```
Of course, this makes no sense and breaks the functionality of this script in all cases.

You can fix this by moving these 2 lines back where they are originally and then also the initialisation is fixed and the error of not having initialised these variables is gone. Besides, they would have been of type `str` and not necessarily a union of `str | None`, but this is beside the point: These two lines are the initialisation of these variables, and changing their indentation into the if-clause handling missing command line arguments removed the initialisation.

Of course, revert the change where the two added lines that initialize these strings to `None` were added after the initial review.

In the context of fixing the indentation around these lines, only the indentation of these two lines needed to be fixed from starting at column 7 to start at column 8:
```
        print("Usage: %s <device> <file>" % sys.argv[0])
        sys.exit(2)
```
No other change is to be done around these lines.

Reply:
my bad , let me fix the indentation  issue .